torch/csrc/distributed/c10d/TCPStore.cpp

```diff
@@ -850,7 +850,9 @@ std::mutex TCPServer::cache_mutex_{};
 
 std::shared_ptr<TCPServer> TCPServer::start(const TCPStoreOptions& opts) {
   auto startCore = [&opts]() {
-    Socket socket = Socket::listen(opts.port);
+    Socket socket = opts.masterListenFd.has_value()
+        ? Socket::listenFromFd(*opts.masterListenFd, opts.port)
```

Comment:
from a debuggability standpoint, it might still be useful to log the `<address:port>` from opts here even in the case we are using an pre-existing FD. In the current path, that `<address:port>` gets logged as part of `Socket::listen(opts.port);`

Reply:
Thank you for the review, @H-Huang! Currently I have an info line to log the fd and port at the end of `SocketListenFromFdOp::run()`:
```
  C10D_INFO(
      "The server has taken over the listening socket with fd {} and port {}",
      fd_,
      expected_port_);
```

I can extend that to also include the address, which I believe is what you are referring to. Please correct me if I'm misunderstanding.